demo/site/src/app/[domain]/[preview]/[language]/[[...path]]/layout.tsx

```diff
@@ -1,19 +1,20 @@
-import { gql, previewParams } from "@comet/cms-site";
-import { GQLLayoutQuery, GQLLayoutQueryVariables } from "@src/app/[domain]/[language]/[[...path]]/layout.generated";
+import { gql } from "@comet/cms-site";
+import { GQLLayoutQuery, GQLLayoutQueryVariables } from "@src/app/[domain]/[preview]/[language]/[[...path]]/layout.generated";
 import { Footer } from "@src/layout/footer/Footer";
 import { footerFragment } from "@src/layout/footer/Footer.fragment";
 import { createGraphQLFetch } from "@src/util/graphQLClient";
-import { getSiteConfigForDomain } from "@src/util/siteConfig";
+import { setPreviewParam } from "@src/util/NotFoundContext";
+import { getSiteConfigForDomain, mapPreviewParamToPreviewData } from "@src/util/siteConfig";
 import type { Metadata } from "next";
 import { PropsWithChildren } from "react";
 
 interface LayoutProps {
-    params: { domain: string; language: string };
+    params: { domain: string; language: string; preview: string };
 }
 
-export default async function Layout({ children, params: { domain, language } }: PropsWithChildren<LayoutProps>) {
-    const { previewData } = (await previewParams()) || { previewData: undefined };
-    const graphqlFetch = createGraphQLFetch(previewData);
+export default async function Layout({ children, params: { domain, language, preview } }: PropsWithChildren<LayoutProps>) {
+    setPreviewParam(preview);
+    const graphqlFetch = createGraphQLFetch(mapPreviewParamToPreviewData(preview));
```

Comment:
- you could instead of `setPreviewParam(preview)` use something like `setPreviewData(mapPreviewParamToPreviewData(preview))` to avoid having to use `mapPreviewParamToPreviewData`everywhere
- you could even put the graphqlFetch into the context: `setGraphQLFetch(createGraphQLFetch(mapPreviewParamToPreviewData(preview)));`